server/src/core/search/search_executor/postprocess.rs

```diff
@@ -200,39 +202,38 @@ fn parse_room_formats(
                 // look nice. Since this building name here serves only search a
                 // hint, we'll crop it (with more from the end, because there
                 // is usually more entropy)
-                (
-                    None,
-                    format!(
-                        "{} {}…{}",
-                        arch_id,
-                        hit.parent_building_names[0].get(..7).unwrap(),
-                        hit.parent_building_names[0]
-                            .get((hit.parent_building_names[0].len() - 10)..)
-                            .unwrap()
-                    ),
-                )
+                let pn = hit.parent_building_names[0].as_str();
+                let (first, _) = pn.unicode_truncate(7);
+                let (last, _) = pn.unicode_truncate_start(10);
+                (None, format!("{arch_id} {first}…{last}"))
             } else {
                 (
                     None,
                     format!("{} {}", arch_id, hit.parent_building_names[0]),
                 )
             }
         };
+        let parsed_aid = unicode_split_at(&parsed_arch_id, search_tokens[0].s.chars().count());
         Some(format!(
             "{}{}{}{}{}",
             prefix.unwrap_or_default(),
             highlighting.0,
-            parsed_arch_id.get(..search_tokens[0].s.len()).unwrap(),
+            parsed_aid.0,
             highlighting.1,
-            parsed_arch_id
-                .get(search_tokens[0].s.len()..)
-                .unwrap_or_default(),
+            parsed_aid.1,
         ))
     } else {
         None
     }
 }
 
+fn unicode_split_at(search: &str, width: usize) -> (String, String) {
+    (
```

Comment:
Hmm… 
The thing I am a bit unhappy with style wise is that we produce 2x`String`, while we only need `&str`
Performance wise this likely does not make a difference, but I am going to try to make this a bit cleaner.